tutorials/generative/2d_controlnet/2d_controlnet.py

```diff
@@ -383,33 +375,53 @@
+                                            timesteps = timesteps,
+                                           cn_cond = masks, )
                     val_loss = F.mse_loss(noise_pred.float(), noise.float())
 
+
             val_epoch_loss += val_loss.item()
+
             progress_bar.set_postfix({"val_loss": val_epoch_loss / (step + 1)})
             break
+
         val_epoch_loss_list.append(val_epoch_loss / (step + 1))
 
         # Sampling image during training with controlnet conditioning
-        progress_bar_sampling = tqdm(scheduler.timesteps, total=len(scheduler.timesteps), ncols=110)
-        progress_bar_sampling.set_description("sampling...")
-        sample = torch.randn((1, 1, 64, 64)).to(device)
-        for t in progress_bar_sampling:
-            with torch.no_grad():
-                with autocast(enabled=True):
-                    down_block_res_samples, mid_block_res_sample = controlnet(
-                        x=sample, timesteps=torch.Tensor((t,)).to(device).long(), controlnet_cond=masks[0, None, ...]
-                    )
-                    noise_pred = model(
-                        sample,
-                        timesteps=torch.Tensor((t,)).to(device),
-                        down_block_additional_residuals=down_block_res_samples,
-                        mid_block_additional_residual=mid_block_res_sample,
-                    )
-                    sample, _ = scheduler.step(model_output=noise_pred, timestep=t, sample=sample)
+
+        with torch.no_grad():
+            with autocast(enabled=True):
+                noise = torch.randn((1, 1, 64, 64)).to(device)
+                sample = controlnet_inferer.sample(
+                    input_noise = noise,
+                    diffusion_model = model,
+                    controlnet = controlnet,
+                    cn_cond = masks[0, None, ...],
+                    scheduler = scheduler,
+                )
+
+        # Without using an inferer:
+#         progress_bar_sampling = tqdm(scheduler.timesteps, total=len(scheduler.timesteps), ncols=110)
+#         progress_bar_sampling.set_description("sampling...")
+#         sample = torch.randn((1, 1, 64, 64)).to(device)
+#         for t in progress_bar_sampling:
+#             with torch.no_grad():
+#                 with autocast(enabled=True):
+#                     down_block_res_samples, mid_block_res_sample = controlnet(
+#                         x=sample, timesteps=torch.Tensor((t,)).to(device).long(), controlnet_cond=masks[0, None, ...]
+#                     )
+#                     noise_pred = model(
+#                         sample,
+#                         timesteps=torch.Tensor((t,)).to(device),
+#                         down_block_additional_residuals=down_block_res_samples,
+#                         mid_block_additional_residual=mid_block_res_sample,
+#                     )
+#                     sample, _ = scheduler.step(model_output=noise_pred, timestep=t, sample=sample)
```

Comment:
Do we want to leave this here or just point people to the definition of the inferer's `sample` method?

Reply:
I was just leaving them there in case a user did not want to use the inferer, as it was previously like that

Reply:
I think its OK to leave it there, it might help users get a bit more insight into what the inferer is actually doing under the hood